compiler/rustc_error_codes/src/error_codes/E0795.md

```diff
@@ -0,0 +1,22 @@
+The `offset_of!` macro gives the offset of a field within a type. It can
+navigate through enum variants, but the final component of its second argument
+must be a field and not a variant.
```

Comment:
This explanation should go after the erroneous code example. Instead it should just be something like:

```suggestion
Invalid argument for the `offset_of!` macro.
```